src/realm/table.hpp

```diff
@@ -1641,6 +1643,19 @@ inline bool Table::has_shared_type() const noexcept
     return !m_top.is_attached();
 }
 
+inline void Table::verify_column(size_t col_ndx, const ColumnBase* col) const
+{
+    if (col_ndx < m_cols.size()) {
+        if (m_cols[col_ndx] == col)
+            return;
+        // The column might be elsewhere in the list
+        for (auto it : m_cols) {
+            if (it == col)
+                return;
+        }
```

Comment:
It's only lucky that `m_cols` is a vector of `ColumnBase*` (pointer primitive) — otherwise this would have invoked a lot of copy constructors. ;-)

I tend to prefer `auto&` out of habit, unless there is a reason to avoid it, but of course using a function from `<algorithms>` avoid all problems completely. ;-)